File: ruby/largest_prime_factor.rb
def largest_prime_factor (x)
    max = 0
    largest_factor = 2
    while Math.sqrt(x).floor() + 1 > largest_factor
        while x % largest_factor == 0
            x = x / largest_factor
            max = largest_factor
        end
        largest_factor += 1
    end
    if x == 1
        return largest_factor
    else
        return x
    end
end

if __FILE__ == $0
    puts largest_prime_factor(600851475143)
end

File: javascript/largest_prime_factor.js
function largest_prime_factor(x) {
    let largest = 2;
    let max = 0;
    while ((Math.floor(Math.sqrt(x)) + 1) > largest) {
        while (x % largest == 0) {
            x /= largest;
            max = largest;
        }
        largest++;
    }
    if (x == 1) {
        return max;
    } else {
        return x;
    }

}

if (require.main === module){
    console.log(largest_prime_factor(600851475143));
}